sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -47,10 +56,56 @@ trait RunnableCommand extends logical.Command {
 }
 
 /**
+ * A trait for classes that can update its metrics of data writing operation.
+ */
+trait MetricUpdater {
```

Comment:
I'd like to call it `trait InsertionCommand extends RunnableCommand`, as we are updating `avgTime`, `numFiles` etc, which is specific to insertion.